<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FeatureProcessingWidget</class>
 <widget class="QWidget" name="FeatureProcessingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>450</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Feature Processing</string>
  </property>
  <layout class="QVBoxLayout" name="mainVerticalLayout">
   <item>
    <widget class="QLabel" name="baseFeaturesLabel">
     <property name="text">
      <string>Available Base Features:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="baseFeatureTableWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string>Feature Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Type</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="transformationsGroupBox">
     <property name="title">
      <string>Transformations for Selected Feature</string>
     </property>
     <layout class="QVBoxLayout" name="transformationsVLayout">
      <item>
       <widget class="QLabel" name="selectedFeatureNameLabel">
        <property name="text">
         <string>Selected: None</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="identityTransformCheckBox">
        <property name="text">
         <string>Use Original (Identity Transformation)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="squaredTransformCheckBox">
        <property name="text">
         <string>Use Squared Value (Squared Transformation)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="separatorTransformations1">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="lagLeadTransformCheckBox">
        <property name="text">
         <string>Use Lag/Lead Features</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="lagLeadParamsGroupBox">
        <property name="title">
         <string>Lag/Lead Parameters</string>
        </property>
        <property name="enabled">
         <bool>false</bool>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="minLagLabel">
           <property name="text">
            <string>Min Lag Steps (-N to 0):</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="minLagSpinBox">
           <property name="minimum">
            <number>-100</number>
           </property>
           <property name="maximum">
            <number>0</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="maxLeadLabel">
           <property name="text">
            <string>Max Lead Steps (0 to +N):</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="maxLeadSpinBox">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="transformationsVerticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="separatorLine">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="zScoreNormalizationCheckBox">
     <property name="text">
      <string>Z-Score Normalize Features (excludes binary features)</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="activeFeaturesDisplayLabel">
     <property name="text">
      <string>Active Features for Model:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="activeFeaturesTableWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::NoSelection</enum>
     </property>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string>Output Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Base Feature</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Transformation</string>
      </property>
     </column>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 